prebuilt-tasks/src/test/java/com/redhat/parodos/tasks/azure/AzureCreateVirtualMachineTaskTest.java

```diff
@@ -0,0 +1,140 @@
+
+public class AzureCreateVirtualMachineTaskTest {
+
+	private static WorkContext ctx;
+
+	private static AzureResourceClient azureResourceClient;
+
+	// Define all Azure resources mocks
+	private static AzureCreateVirtualMachineTask underTest;
+
+	private static ResourceGroup resourceGroup;
+
+	private static AvailabilitySet availabilitySet;
+
+	private static PublicIpAddress publicIPAddress;
+
+	private static Network network;
+
+	private static NetworkInterface networkInterface;
+
+	private static VirtualMachine virtualMachine;
+
+	private static final String[] requiredParamsKeys = { AzureCreateVirtualMachineTask.VM_USER_NAME_KEY,
+			AzureCreateVirtualMachineTask.VM_SSH_PUBLIC_KEY_KEY, AzureCreateVirtualMachineTask.AZURE_TENANT_ID_KEY,
+			AzureCreateVirtualMachineTask.AZURE_SUBSCRIPTION_ID_KEY, AzureCreateVirtualMachineTask.AZURE_CLIENT_ID_KEY,
+			AzureCreateVirtualMachineTask.AZURE_CLIENT_SECRET_KEY,
+			AzureCreateVirtualMachineTask.AZURE_RESOURCES_PREFIX_KEY };
+
+	private static final String resourcesPrefix = AzureCreateVirtualMachineTask.AZURE_RESOURCES_PREFIX_KEY
+			+ "-testValue";
+
+	private static final String userName = AzureCreateVirtualMachineTask.VM_USER_NAME_KEY + "-testValue";
+
+	private static final String sshKey = AzureCreateVirtualMachineTask.VM_SSH_PUBLIC_KEY_KEY + "-testValue";
+
+	@BeforeEach
+	public void setUp() throws Exception {
+		// Initiate all Azure resources mocks
+		azureResourceClient = mock(AzureResourceClient.class);
+		resourceGroup = mock(ResourceGroup.class);
+		availabilitySet = mock(AvailabilitySet.class);
+		publicIPAddress = mock(PublicIpAddress.class);
+		network = mock(Network.class);
+		networkInterface = mock(NetworkInterface.class);
+		virtualMachine = mock(VirtualMachine.class);
+
+		underTest = new AzureCreateVirtualMachineTask(azureResourceClient);
+		underTest.setBeanName("AzureCreateVirtualMachineTask");
+		ctx = new WorkContext();
+
+		HashMap<String, String> map = new HashMap<>();
+		for (String paramKey : requiredParamsKeys) {
+			map.put(paramKey, paramKey + "-testValue");
+			WorkContextDelegate.write(ctx, WorkContextDelegate.ProcessType.WORKFLOW_TASK_EXECUTION, underTest.getName(),
+					WorkContextDelegate.Resource.ARGUMENTS, map);
+		}
+
+		when(azureResourceClient.createResourceGroup(resourcesPrefix)).thenReturn(resourceGroup);
+		when(azureResourceClient.createAvailabilitySet(resourcesPrefix)).thenReturn(availabilitySet);
+		when(azureResourceClient.createPublicIpAddress(resourcesPrefix)).thenReturn(publicIPAddress);
+		when(azureResourceClient.createNetwork(resourcesPrefix)).thenReturn(network);
+		when(azureResourceClient.createNetworkInterface(resourcesPrefix, network, publicIPAddress))
+				.thenReturn(networkInterface);
+		when(azureResourceClient.createVirtualMachine(resourcesPrefix, networkInterface, availabilitySet, userName,
+				sshKey)).thenReturn(virtualMachine);
+	}
+
+	@Test
+	public void testHappyFlow() {
+		String publicIP = "11.11.11.11";
```

Comment:
pls add tests using the convention:
```
/ /given
...
// when
..
// then
..
```
see as en example https://github.com/parodos-dev/parodos/blob/main/workflow-service/src/test/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImplTest.java#L175

Reply:
Done